public/js/app.js

```diff
@@ -17,17 +18,35 @@ var setScene = function() {
 
   var setSceneCall = new XMLHttpRequest();
   setSceneCall.open("PUT", `/room/${roomId}/scene/${sceneId}`, true);
+  setSceneCall.onreadystatechange = updateState;
   setSceneCall.send();
 };
 
 for (var i = 0; i < checkboxes.length; i++) {
   checkboxes[i].addEventListener('change', onOff, false);
 }
 
-for (var i = 0; i < rooms.length; i++) {
-  rooms[i].style.backgroundColor = rooms[i].getAttribute("data-colour");
+for (var i = 0; i < sceneImages.length; i++) {
+  sceneImages[i].addEventListener('click', setScene, false);
 }
 
-for (var i = 0; i < scenes.length; i++) {
-  scenes[i].addEventListener('click', setScene, false);
+var updateState = function() {
+  var getStateCall = new XMLHttpRequest();
+  getStateCall.open("GET", "/groups/state", true);
+  getStateCall.onreadystatechange = function() {
+    if (this.readyState == 4 && this.status == 200) {
```

Comment:
I wonder if there's a constant (that evaluates to 4) to make this more readable?

Reply:
Oh yeah, there is! `XMLHttpRequest.DONE` 😃